Fix/unreadable device

szymonlesisz opened the pull request:
Distinguish unreadable device from unacquired device.
Unreadable device error (reason) is passed in unreadable Device.id field.

Device selection is required if unreadable device is present (popup mode)

matejkriz commented on src/js/env/browser/browserUtils.js:
```diff
@@ -51,6 +51,12 @@ export const getBrowserState = (supportedBrowsers: {
+export const getOS = () => {
+    if (!navigator || !navigator.userAgent) return 'unknown';
+    const browser = Bowser.getParser(navigator.userAgent);
+    return browser.getOS().name.toLowerCase();
```
`name` is defined as possibly undefined (probably could never happen, but better safe than sorry)

szymonlesisz replied:
addressed in 4270cee

also a lot of tests were added

szymonlesisz committed:
fix: add unreadable Device error details

Handle any LIBUSB_ERROR from trezord.
Distinguish unreadable device from unacquired device.
Unreadable device error (reason) is passed in unreadable Device.error field and could be processed by UI.

szymonlesisz committed:
fix: show unreadable device in popup

refactored device selection process in Core since unreadable device is no longer webusb specific behaviour.
change explanation in popup view depending on unreadable device error.

szymonlesisz committed:
chore: refactor browserUtils

unify usage of Bowser (parse/getParser), catch possible undefined values.
added unit tests.

matejkriz commented on src/ts/types/__tests__/index.ts:
```diff
@@ -87,6 +87,13 @@ export const events = async () => {
             payload.features.unfinished_backup;
             payload.features.no_backup;
             payload.features.model;
+            // @ts-ignore: error does not exist
+            payload.error.toLowerCase();
```
maybe this would work in TS file?
```suggestions
payload.error?.toLowerCase();
```

szymonlesisz replied:
`@ts-expect-error` it is, thanks

szymonlesisz committed:
fix: bootloader T1 getFeatures timeout handling
